core/src/main/scala/kafka/log/Log.scala

```diff
@@ -301,7 +303,13 @@ class Log(@volatile var dir: File,
   private def initializeLeaderEpochCache(): LeaderEpochFileCache = {
     // create the log directory if it doesn't exist
     Files.createDirectories(dir.toPath)
-    val checkpointFile = new LeaderEpochCheckpointFile(LeaderEpochFile.newFile(dir), logDirFailureChannel)
+    val file = LeaderEpochFile.newFile(dir)
+    if (!supportsLeaderEpoch && file.exists()) {
```

Comment:
We delete the file, but then immediately recreate it. That means we will see the warning every time we initialize the cache. Rather than deleting, I wonder if we could check if the existing cache is empty, and if so, log a warning and call `clearAndFlush`.